### PR TITLE
AP_Periph: fix network build

### PR DESCRIPTION
**Pull request by magicrub**
fix Periph network builds

### CODE REVIEW

File: Tools/AP_Periph/networking.h [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #pragma once
 
-#include "AP_Periph.h"
-
 #ifdef HAL_PERIPH_ENABLE_NETWORKING
```
Comment (peterbarker) [inline Tools/AP_Periph/networking.h:3]: We need to include *something* before this or we're dependent on include order whereever this is included.

File: libraries/AP_Networking/AP_Networking_port.cpp [hunk 2]
```diff
@@ -353,9 +353,11 @@ bool AP_Networking::Port::send_receive(void)
         // handle outgoing packets
         uint32_t available = writebuffer->available();
         available = MIN(300U, available);
+#if HAL_GCS_ENABLED
```
Comment (peterbarker) [inline libraries/AP_Networking/AP_Networking_port.cpp:356]: This'll do.

I should probably try to get my `AP_MAVLINK_ENABLED` branch working at some stage...

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: Tools/AP_Periph/networking.h (hunk 1 above)
Comment (magicrub) [reply Tools/AP_Periph/networking.h:5]: yup. Added <AP_HAL/AP_HAL_Boards.h>

**Review by peterbarker — APPROVED** [review]
LGTM

**Pull request merged**